runtime/runtime-params-estimator/src/qemu.rs

```diff
@@ -39,3 +41,72 @@ fn hypercall(index: u32) -> u64 {
     }
     result
 }
+
+/// Create a command to be executed inside QEMU with the custom counter plugin.
+pub struct QemuCommandBuilder {
+    started: bool,
+    on_every_close: bool,
+    count_per_thread: bool,
+    plugin_log: bool,
+}
+
+impl QemuCommandBuilder {
+    /// Start measurement immediately, without having to call `start_count_instructions` first.
+    pub fn started(mut self, started: bool) -> Self {
+        self.started = started;
+        self
+    }
+    /// Print the counters on every close() syscall
+    pub fn print_on_every_close(mut self, on_every_close: bool) -> Self {
+        self.on_every_close = on_every_close;
+        self
+    }
+    /// Instantiate different counters for each thread
+    pub fn count_per_thread(mut self, count_per_thread: bool) -> Self {
```

Comment:
I like the following naming convention for such setters (seen in some of BurntSushi's code)

```suggestion
    pub fn count_per_thread(mut self, yes: bool) -> Self {
```